packages/next-auth/src/providers/discord.ts

```diff
@@ -1,5 +1,26 @@
-/** @type {import(".").OAuthProvider} */
-export default function Discord(options) {
+import type { OAuthConfig, OAuthUserConfig } from "."
+
+export interface DiscordProfile {
+  accent_color: number
+  avatar: string
+  banner: string
+  banner_color: string
+  discriminator: string
+  email: string
+  flags: number
+  id: string
+  image_url: string
+  locale: string
+  mfa_enabled: boolean
+  premium_type: number
+  public_flags: number
+  username: string
+  verified: boolean
+}
+
+export default function Discord<P extends DiscordProfile>(
```

Comment:
I think the safest way to type this would be:
```suggestion
export default function Discord<P extends Record<string, any> = DiscordProfile>(
```
It's the pattern we use in other providers as well 👍🏽

Reply:
Hey @lluia! Thanks for the suggestion -- perhaps you can provide some guidance here:

I attempted to provide this typing, but it actually causing a typing error on lines 14 and 17 (35 and 38 in the new code) where `profile.image_url` has its value set. Honestly, the error is a bit confusing because the typing should be correct.

Any idea what the reasoning is? Note that it's fine on line 23 (44 in the new code).

Reply:
@lluia Good catch! I've made the adjustments as requested.

If I find time this week, I'll look into adjusting and testing the other providers' interfaces as well to have the correct typing. Appreciate your help!

Reply:
Going to mark this as resolved, let me know if you need anything else!